catalog/spatial/geocoding/spatial-geocoding-extract/src/main/java/org/codice/ddf/spatial/geocoding/extract/GeoNamesFileExtractor.java

```diff
@@ -248,10 +257,14 @@ private InputStream getInputStreamFromUrl(
   }
 
   public Response createConnection(String resource) {
-    webClient = WebClient.create(url + resource + ".zip");
-    webClient.path(url + resource + ".zip");
-    Response response = webClient.get();
-    return response;
+    if (urlValidator.isValid(resource)) {
```

Comment:
❓ It looks like this method is mocked in the unit tests and never gets tested. Does the urlValidator need to be tested here?

Reply:
Yeah there are two methods that are overridden by the `Spy` from what I can see. Both `getUrlInputStreamFromWebClient` and `createConnection` are never touched by any of the exisiting tests. This class would need to be refactored a bit to make it easier to test it. I think one of the main problems I noticed is that there is no way to provide a mock `WebClient` since the `createConnection` method makes a call to `WebClient.create`. It looks like @pvargas followed the pattern of the existing tests here. Should he refactor this class a bit to make it possible to test everything?

Reply:
I think manual testing might be sufficient in this case.